ttt/minimax.rb

```diff
@@ -0,0 +1,75 @@
+require_relative 'ai'
+require_relative 'symbols'
+
+class Minimax
+  @board
+  @ai_symbol
+  @current_symbol
+  @human_symbol
+  @winner
+  @symbols
+
+  def initialize(board, symbols)
+    @board = board
+    @symbols = symbols
+    @ai_symbol = symbols.get_ai
+    @human_symbol = symbols.get_human
+  end
+
+  def get_best_move
+    depth = 1
+    best_move = 0
+    best_score = -100
+
+    original_board = @board.current_state
+    copy_of_board = [].replace(original_board)
+    @current_symbol = @ai_symbol
+
+    @board.find_all_empty_positions.each do | position |
+      copy_of_board[position] = @current_symbol
+      score = compute(copy_of_board, depth += 1 )
+      if score > best_score
+        best_score = score
+        best_move = position
+      end
+      copy_of_board = [].replace(original_board)
```

Comment:
You may be able to use `dup` or `clone` to create a copy, too.

https://apidock.com/ruby/Object/dup#1143-Dup-vs-Clone-difference